### ISSUE TITLE
azurerm_monitor_diagnostic_setting doesn't accept 'log' anymore: `"enabled_log": one of `enabled_log,metric` must be specified`

### ISSUE THREAD

**Issue by asbiin** [hashicorp/terraform-provider-azurerm#23180]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.5.4

### AzureRM Provider Version

3.71.0

### Affected Resource(s)/Data Source(s)

azurerm_monitor_diagnostic_setting

### Terraform Configuration Files

```hcl
resource "azurerm_resource_group" "this" {
  name     = "diagnostic-test"
  location = "westeurope"
}

data "azurerm_client_config" "current" {}

resource "azurerm_key_vault" "this" {
  name                       = "<keyvault name>"
  location                   = azurerm_resource_group.this.location
  resource_group_name        = azurerm_resource_group.this.name
  tenant_id                  = data.azurerm_client_config.current.tenant_id
  soft_delete_retention_days = 7
  purge_protection_enabled   = false
  sku_name                   = "standard"
}

resource "azurerm_monitor_diagnostic_setting" "logs" {
  name               = "diagnostic-test-log"
  target_resource_id = azurerm_key_vault.this.id

  log_analytics_workspace_id = azurerm_log_analytics_workspace.this.id

  log {
    category = "AuditEvent"
  }
}

resource "azurerm_log_analytics_workspace" "this" {
  name                = "diagnostic-test-log"
  resource_group_name = azurerm_resource_group.this.name
  location            = azurerm_resource_group.this.location
  sku                 = "PerGB2018"
  retention_in_days   = 30
}

terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "~> 3.71.0"
    }
  }
}

provider "azurerm" {
  features {}
}
```

### Debug Output/Panic Output

```shell
https://gist.github.com/asbiin/eb9a3499882e7e7d4040fe333e1c8f18
```

### Expected Behaviour

The `log` property should still be accepted by `azurerm_monitor_diagnostic_setting`, although there is a warning message about deprecation:

```
│ Warning: Argument is deprecated
│
│   with azurerm_monitor_diagnostic_setting.logs,
│   on main.tf line 18, in resource "azurerm_monitor_diagnostic_setting" "logs":
│    6: resource "azurerm_monitor_diagnostic_setting" "logs" {
│
│ `log` has been superseded by `enabled_log` and will be removed in version 4.0 of the AzureRM Provider.
```

### Actual Behaviour

Adding only `log` property is not working anymore, an error message appears, although the deprecation message says it will be removed on 4.0 version only.

```
│ Error: Missing required argument
│
│   with azurerm_monitor_diagnostic_setting.logs,
│   on main.tf line 18, in resource "azurerm_monitor_diagnostic_setting" "logs":
│    6: resource "azurerm_monitor_diagnostic_setting" "logs" {
│
│ "enabled_log": one of `enabled_log,metric` must be specified
```

### Steps to Reproduce

1. terraform init
2. terraform apply

**Comment by teowa** [comment]
Hi @asbiin , this is fixed in PR https://github.com/hashicorp/terraform-provider-azurerm/pull/23093 and will be released in [3.72.0](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/CHANGELOG.md#3720-unreleased) version of the provider.

**Comment by rcskosir** [comment]
@asbiin Thank you for taking the time to open this issue, as noted above by @teowa, this is fixed in PR https://github.com/hashicorp/terraform-provider-azurerm/pull/23093 and now has been released in [3.72.0](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/CHANGELOG.md#3720-unreleased) version of the provider, so I am going to go ahead and close this issue. Thanks again.
